### PR TITLE
ENG-17757 make the update of last heart beat  time thread safe

### PR DESCRIPTION
**Pull request by zeemanhe**

### CODE REVIEW

File: src/frontend/org/voltcore/messaging/ForeignHost.java [hunk 1]
```diff
@@ -266,12 +267,11 @@ private void detectDeadHost() {
...
         // set m_isUp to false, so use both that and m_closing to
         // avoid repeat reports of a single node failure
-        if (m_isUp && current_delta > m_deadHostTimeout)
-        {
+        if (m_isUp && current_delta > m_deadHostTimeout) {
             if (m_deadReportsCount.getAndIncrement() == 0) {
                 hostLog.error("DEAD HOST DETECTED, hostname: " + hostnameAndIPAndPort());
                 hostLog.info("\tcurrent time: " + current_time);
-                hostLog.info("\tlast message: " + m_lastMessageMillis);
+                hostLog.info("\tlast message: " + m_lastMessageMillis.get());
```
Comment (jmcgrail-voltdb) [inline src/frontend/org/voltcore/messaging/ForeignHost.java]: this may print a different value than was used in the calculation

**Comment by luyangco** [comment]
Given that it's used in pico-network's read path, does it worth to make the `m_lastMessageMillis` thread-safe? Comparing to our default  10s timeout setting, the deviated time difference would be a few milliseconds at most.

**Comment by n3world** [comment]
I'm not sure if it is important but this change does not actual change how this class behaves other than wrapping the long in an Object

**Comment by zeemanhe** [comment]
In one of the tests, the current and last time were the same but its delta was beyond the configured timeout, which triggered "DEAD HOST". This report should be a false alarm. Ran a couple of rounds of perf benchmarks on this branch,  did not see degradation. Will double check perf on master branch.

**Pull request merged**